src/Symfony/Component/Finder/Tests/BsdFinderTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Finder\Tests;
+
+use Symfony\Component\Finder\Adapter\BsdFindAdapter;
+
+/**
+ * @group legacy
+ */
+class BsdFinderTest extends AbstractFinderTest
+{
+    protected function getValidAdapters()
+    {
+        $adapter = new BsdFindAdapter();
+
+        if (!$adapter->isSupported()) {
+            $this->markTestSkipped(get_class($this).' is not supported.');
+        }
+
+        return array($adapter);
```

Comment:
Instead of a method returning an array of adapters with a single element, and then using this to build a dataProvider providing the adapter, I suggest you to use a protected method ``getAdapter`` returning the adapter being used for tests and using it in ``buildFinder()`` (which would not need an argument anymore). It would make tests much easier to understand